### PR TITLE
Converted Notes store to Vue 3.0

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15705](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15705)

## Description

Converted Notes store to Vue 3.0

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    // Helpers
+    function handleError(
+        hdid: string,
+        resultError: ResultError,
+        errorType: ErrorType
+    ): void {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        logger.error(`ERROR: ${JSON.stringify(resultError)}`);
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:85]:
```suggestion
        logger.error(`ERROR: ${JSON.stringify(resultError)}`);
```

The logger has already been retrieved.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        statusMessage.value = resultError.resultMessage;
+        status.value = LoadStatus.ERROR;
+    }
+
+    // Helpers
+    function handleError(
+        hdid: string,
+        resultError: ResultError,
+        errorType: ErrorType
+    ): void {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:82]:
```suggestion
    function handleError(
        resultError: ResultError,
        errorType: ErrorType
    ): void {
```

This function didn't have an HDID parameter previously.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                resultError.traceId
+            );
+        }
+    }
+
+    // Actions
+    function retrieveNotes(params: { hdid: string }): Promise<void> {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        const userNotes: UserNote[] = notes.value;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:109]:
```suggestion
        const userNotes: UserNote[] = notes.value;
```

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+    function createNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<UserNote | undefined> {
+        return noteService
+            .createNote(params.hdid, params.note)
+            .then((result) => {
+                if (result !== undefined) {
+                    setCreatedNote(result);
```
[the remark this answers is not in the document]
Comment (BronzBierd) [reply Apps/WebClient/src/NewClientApp/src/stores/note.ts:145]: @MikeLyttle 

I haven't followed this myself, but I'm thinking of naming that relies on a set of mutation verbs:

1. `set**` for changing singular changes, so `setUpdateNote` would be appropriate but not for adding a note to the collection.
2. `add**` for extending a collection
3. `clear**` for removing one or all e.g. clearStorage, clearNotification, clearNotifications

For actions we would rely on more conventional verbs:
1. `create**`
2. `update**`
3. `edit**`
4. `delete**`

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    function deleteNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<void> {
+        return noteService
+            .deleteNote(params.hdid, params.note)
+            .then(() => {
+                setDeletedNote(params.note);
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:180]: depending on the convention comment, it would be better to use `clearDeletedNote`

**Review by BronzBierd — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+            errorStore.addError(
+                errorType,
+                ErrorSourceType.Note,
+                resultError.traceId
+            );
+        }
+    }
+
+    // Actions
+    function retrieveNotes(params: { hdid: string }): Promise<void> {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:106]: The signature should be simplified to `retrieveNotes(hdid: string)`.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+                    handleError(params.hdid, error, ErrorType.Retrieve);
+                    throw error;
+                });
+        }
+    }
+
+    function createNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<UserNote | undefined> {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:140]: The signature should be simplified to `createNote(hdid: string, note: UserNote)`.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            .catch((error: ResultError) => {
+                handleError(params.hdid, error, ErrorType.Create);
+                throw error;
+            });
+    }
+
+    function updateNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<UserNote> {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:160]: The signature should be simplified to `updateNote(hdid: string, note: UserNote)`.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+            .catch((error: ResultError) => {
+                handleError(params.hdid, error, ErrorType.Update);
+                throw error;
+            });
+    }
+
+    function deleteNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<void> {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/note.ts:176]: The signature should be simplified to `deleteNote(hdid: string, note: UserNote)`.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts (hunk 5 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/note.ts:177]: Used @MikeLyttle recommended name instead.

File: Apps/WebClient/src/NewClientApp/src/stores/note.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/note.ts:142]: Used @MikeLyttle suggestions for "mutation" names.

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**